### ISSUE TITLE
citekey field in bibtex format should not contain blank spaces or commas

### ISSUE THREAD

**Issue by ebergasa** [michel-kraemer/citeproc-java#217]
**Describe the bug**
While exporting to 'bibtex' format, citekey field is built by concatenating author names with underscores.

The problem occurs when there are blanks in family names. Blank spaces should be replaced by underscores as well.

**Which version of citeproc-java are you using?**
3.0.0-beta.2

**Expected behavior**
bibtex citekey field should not contain blanks or commas.

As of [BibTex format standard](https://www.bibtex.com/g/bibtex-format/):
The citekey can be any combination of alphanumeric characters including the characters "-", "_", and ":"

**Used style and locale**
bibtex es-ES

**Example code/unit test**

```
    public static void main(String[] args) throws IOException {
        var item = new CSLItemDataBuilder().id("6501a713df0d06753d5bf43f")
                .type(CSLType.BOOK)
                .language("es")
                .author(    new CSLNameBuilder()
                            .family("Raya Díez")
                                .given("Esther")
                            .build(),
                        new CSLNameBuilder()
                                .literal("Arruga Serrano, Concepción")
                                .build()
                        )
                .issued(2021)
                .title("Brecha digital y competencias digitales: estudio de la situación en La Rioja")
                .build();
        
        var itemDataProvider = new ListItemDataProvider(List.of(item));
        
        var csl = new CSL(itemDataProvider, "bibtex", "es-ES");
        
        csl.registerCitationItems(itemDataProvider.getIds());
        csl.setOutputFormat("text");
        
```
Results in:

` @book{raya díez_arruga serrano, concepción_2021, title={Brecha digital y competencias digitales: estudio de la situación en La Rioja}, author={Raya Díez, Esther and Arruga Serrano, Concepción}, year={2021} }
`

**Comment by ebergasa** [comment]
It seems that this is not a real issue with citeproc-java. This issue has already been reported in CSL https://github.com/citation-style-language/styles/issues/4753

**Comment by ebergasa** [comment]
To allow improvements in bibtex.csl and be able to fix this it might be necessary that citeproc-java supported new string variable 'citation-key', present in [CSL 1.0.2] (https://github.com/citation-style-language/schema/releases/tag/v1.0.2) 

Do you find it feasible @michel-kraemer  ?

Thanks!

**Comment by michel-kraemer** [comment]
Thanks for reporting this. Unfortunately, this cannot be fixed at the moment since no style currently supports the new `citation-key` variable. See for example `bibtex.csl`:
https://github.com/citation-style-language/styles/blob/master/bibtex.csl

Even if I added this variable, it would not be used. I'll keep this issue open and add the variable once the `bibtex` style gets updated.

**Comment by michel-kraemer** [comment]
I've received another request to implement the `citation-key` variable, so I've decided to do it now. I've just released version 3.1.0. Have fun!
